js/modules/loki_app_dot_net_api.js

```diff
@@ -717,104 +754,220 @@ class LokiPublicChannelAPI {
-          const sameUsername = message.username === adnMessage.user.username;
-          const sameText = message.text === adnMessage.text;
-          // Don't filter out messages that are too far apart from each other
-          const timestampsSimilar =
-            Math.abs(message.timestamp - timestamp) <=
-            PUBLICCHAT_MIN_TIME_BETWEEN_DUPLICATE_MESSAGES;
-
-          return sameUsername && sameText && timestampsSimilar;
-        };
+          // Duplicate check
+          const isDuplicate = message => {
+            // The username in this case is the users pubKey
+            const sameUsername = message.username === adnMessage.user.username;
+            const sameText = message.text === adnMessage.text;
+            // Don't filter out messages that are too far apart from each other
+            const timestampsSimilar =
+              Math.abs(message.timestamp - timestamp) <=
+              PUBLICCHAT_MIN_TIME_BETWEEN_DUPLICATE_MESSAGES;
+
+            return sameUsername && sameText && timestampsSimilar;
+          };
 
-        // Filter out any messages that we got previously
-        if (this.lastMessagesCache.some(isDuplicate)) {
-          return; // Duplicate message
-        }
+          // Filter out any messages that we got previously
+          if (this.lastMessagesCache.some(isDuplicate)) {
+            return false; // Duplicate message
+          }
 
-        // Add the message to the lastMessage cache and keep the last 5 recent messages
-        this.lastMessagesCache = [
-          ...this.lastMessagesCache,
-          {
-            username: adnMessage.user.username,
-            text: adnMessage.text,
-            timestamp,
-          },
-        ].splice(-5);
-
-        const from = adnMessage.user.name; // profileName
-
-        const messageData = {
-          serverId: adnMessage.id,
-          clientVerified: true,
-          friendRequest: false,
-          source: adnMessage.user.username,
-          sourceDevice: 1,
-          timestamp,
-          serverTimestamp: timestamp,
-          receivedAt,
-          isPublic: true,
-          message: {
-            body: adnMessage.text,
-            attachments: [],
-            group: {
-              id: this.conversationId,
-              type: textsecure.protobuf.GroupContext.Type.DELIVER,
+          // FIXME: maybe move after the de-multidev-decode
+          // Add the message to the lastMessage cache and keep the last 5 recent messages
+          this.lastMessagesCache = [
+            ...this.lastMessagesCache,
+            {
+              username: adnMessage.user.username,
+              text: adnMessage.text,
+              timestamp,
             },
-            flags: 0,
-            expireTimer: 0,
-            profileKey: null,
+          ].splice(-5);
+
+          const from = adnMessage.user.name; // profileName
+          if (pubKeys.indexOf(`@${adnMessage.user.username}`) === -1) {
```

Comment:
use `pubKeys.includes()` ?

Reply:
I stand behind my use of indexOf for the reasons I've posted in messenger channel a couple times

Reply:
alrighty then